### PR TITLE
Bug 1411549 again - Operation Shrink NotificationService

### PR DESCRIPTION
**Pull request by justindarc**

### CODE REVIEW

File: Providers/Profile.swift [hunk 1]
```diff
@@ -715,14 +713,18 @@ open class BrowserProfile: Profile {
...
-                    let url = "https://mozilla-anonymous-sync-metrics.moo.mx/post/bookmarkvalidation".asURL!
+                    guard let url = URL(string: "https://mozilla-anonymous-sync-metrics.moo.mx/post/bookmarkvalidation") else {
+                        return
+                    }
+
                     var request = URLRequest(url: url)
                     request.httpMethod = "POST"
                     request.httpBody = body
-                    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+                    request.addValue("application/json", forHTTPHeaderField: "Content-Type")
```
Comment (jhugman) [inline Providers/Profile.swift]: Why the change?

File: Providers/Profile.swift [hunk 2]
```diff
@@ -2,8 +2,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import Alamofire
-import Foundation
```
Comment (jhugman) [inline Providers/Profile.swift]: You should probably add a warning to future travellers that adding extra imports here may have deleterious effects on push notifications.

File: Extensions/NotificationService/NotificationService.swift [hunk 3]
```diff
@@ -20,46 +17,37 @@ class NotificationService: UNNotificationServiceExtension {
     // AppDelegate.application(_:didReceiveRemoteNotification:completionHandler:)
     // Once the notification is tapped, then the same userInfo is passed to the same method in the AppDelegate.
     override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        DispatchQueue.global().async {
```
Comment (jhugman) [inline Extensions/NotificationService/NotificationService.swift]: Wait, what? I'm not even sure why we're using `Dispatch.main.async` in the first place.

File: Providers/Profile.swift (hunk 1 above)
Comment (justindarc) [reply Providers/Profile.swift]: hm.. likely an oversight.. i copied in the non-alamofire HTTP request code from telemetry-ios
